### PR TITLE
Fix: Resolve date alignment issues and worldTime edge cases (Issue #66)

### PR DESCRIPTION
**Pull request by rayners**
## Summary
- Fixed intercalary days protection with null safety checks for month name access
- Normalized JavaScript precision issues (-0 vs +0) in worldTimeToDate method
- Enhanced PF2e system mock for compatibility testing
- Converted all branching logic in tests to separate test cases
- Made all test assertions exact values instead of ranges
- Replaced random stress test with deterministic round-trip tests

## Test plan
- [x] All unit tests pass (647/647)
- [x] Lint validation complete
- [x] TypeScript compilation successful
- [x] PF2e date alignment edge cases covered
- [x] WorldTime interpretation modes tested
- [x] Calendar-specific boundary conditions validated

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: TEST-FAILURE-BASELINE.md
Comment (rayners) [inline TEST-FAILURE-BASELINE.md:1]: This file doesn't need to be in the pull request.

File: src/core/calendar-engine.ts [hunk 1]
```diff
@@ -339,9 +344,10 @@ export class CalendarEngine {
       remainingDays -= monthLength;
 
       // Check for intercalary days after this month
-      const intercalaryAfterMonth = intercalaryDays.filter(
-        i => i.after === this.calendar.months[month - 1].name
-      );
+      const currentMonthName = this.calendar.months[month - 1]?.name;
+      const intercalaryAfterMonth = currentMonthName
+        ? intercalaryDays.filter(i => i.after === currentMonthName)
+        : [];
```
Comment (rayners) [inline src/core/calendar-engine.ts:350]: These need to be covered by tests.

File: src/core/calendar-engine.ts [hunk 2]
```diff
@@ -402,9 +408,10 @@ export class CalendarEngine {
       totalDays += monthLengths[month - 1];
 
       // Add intercalary days after this month
-      const intercalaryAfterMonth = intercalaryDays.filter(
-        i => i.after === this.calendar.months[month - 1].name
-      );
+      const currentMonthName = this.calendar.months[month - 1]?.name;
+      const intercalaryAfterMonth = currentMonthName
+        ? intercalaryDays.filter(i => i.after === currentMonthName)
+        : [];
```
Comment (rayners) [inline src/core/calendar-engine.ts:414]: Tests for these too.

File: test/calendar-specific-alignment-issues.test.ts
Comment (rayners) [inline test/calendar-specific-alignment-issues.test.ts:1]: Break this up by calendar file. A given test file should depend on no more than one single calendar JSON file. In fact, I think it'd be worth putting them into their own directories. When we transition to externally defined calendars, we could potentially include tests along with the JSON definitions.
